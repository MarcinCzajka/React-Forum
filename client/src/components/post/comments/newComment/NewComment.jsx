import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import { Comment, Button } from 'semantic-ui-react';
import AvatarPlaceholder from '../../../placeholders/AvatarPlaceholder';
import { LocaleConsumer } from '../../../../contexts/LocaleContext';
import './NewComment.scss';

class NewComment extends React.Component {
    constructor(props) {
        super(props);

        this.state = {avatarReady: false, showButtonContainer: false};
    }

    handleAvatarLoaded = () => {
        this.setState({avatarReady: true});
    }

    showButtonContainer = () => {
        this.setState({showButtonContainer: true});
    }

    handleCancelClick = () => {
        this.props.handleReplyChange('');
        this.setState({showButtonContainer: false});
    }

    render() {
        const {avatarReady, showButtonContainer } = this.state;

        return (
            <LocaleConsumer>
                {locale => (
                    <div style={{display: this.props.active ? '' : 'none'}} className="ui large comments newCommentContainer">

                        <Comment className="comment">

                            {!avatarReady ? (
                                <AvatarPlaceholder size='sizePostComment' /> 
                            ) : ''}

                            <Comment.Avatar 
                                style={{display:(avatarReady ? 'block' : 'none')}}
                                className="avatar" 
                                src={this.props.userAvatar}
                                onLoad={this.handleAvatarLoaded}>
                            </Comment.Avatar>
                            <div className='pseudoLoggedIn'></div>


                            <textarea 
                                className={showButtonContainer ? 'active' : ''}
                                placeholder={locale.postWithComment.textareaPlaceholder}
                                value={this.props.replyContent}
                                onChange={({target}) => this.props.handleReplyChange(target.value)}
                                onFocus={this.showButtonContainer}>
                            </textarea>


                            <div className={classNames({buttonContainer: true, hiddenButtonContainer: !showButtonContainer})}>

                                <Button 
                                    compact
                                    floated='right'
                                    color='teal'
                                    disabled={!this.props.replyContent}
                                    content={locale.postWithComment.commentButton} 
                                    onClick={this.props.handleReplyToPost}
                                />

                                <Button 
                                    compact
                                    floated='right'
                                    basic
                                    content={locale.postWithComment.cancelButton} 
                                    onClick={this.handleCancelClick}
                                />

                            </div>


                        </Comment>
                    </div>
                )}
            </LocaleConsumer>
        )
    }

}

NewComment.defaultProps = {
    active: true,
    replyContent: ''
}

NewComment.propTypes = {
    active: PropTypes.bool,
    userAvatar: PropTypes.string,
    replyContent: PropTypes.string,
    handleReplyChange: PropTypes.func.isRequired,
    handleReplyToPost: PropTypes.func.isRequired
}

export default NewComment;