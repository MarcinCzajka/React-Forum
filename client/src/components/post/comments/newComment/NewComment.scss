.newCommentContainer {
    max-width: 100% !important;

    textarea {
        resize: none;
        width: calc(100% - 5em);
        border: 1px solid #ddd;
        border-radius: 10px;
        height: 3em;
        padding: 0.6em;

        transition: border 0.2s ease-out, border-radius 0.2s ease-out;

        &.active {
            border-radius: 0;
            border-bottom: 1px solid #222;
        }
    }

    .pseudoLoggedIn {
        width: 9px;
        height: 9px;
        border-radius: 5px;
        position: absolute;
        background-color: #95d731;
        left: 2.2em;
        top: 2.2em;
    }

    .buttonContainer {
        padding: 0.5em 2.5em 1em 0;
        transition: padding 0.2s ease-out;

        &.hiddenButtonContainer {
            padding: 0.5em 2.5em 0 0;
            visibility: hidden;
        }
    }
}