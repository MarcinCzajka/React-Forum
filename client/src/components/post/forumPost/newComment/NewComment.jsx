import React from 'react';
import PropTypes from 'prop-types';
import { Form, Button } from 'semantic-ui-react';

const NewComment = props => {
    if(props.active) {
        return (
            <Form reply style={{marginTop: '0.5em'}}>
                <Form.TextArea 
                    value={props.replyContent}
                    onChange={props.handleReplyChange} />
            </Form>
        ) 
    } else {
        return '';
    }
}

NewComment.defaultProps = {
    replyContent: '',
    active: false
}

NewComment.propTypes = {
    replyContent: PropTypes.string,
    active: PropTypes.bool,
    handleReplyChange: PropTypes.func.isRequired,
    handleReplyToPost: PropTypes.func.isRequired
}

export default NewComment;