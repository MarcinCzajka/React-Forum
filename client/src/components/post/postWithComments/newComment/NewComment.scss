.newCommentContainer {
    max-width: 100% !important;

    textarea {
        resize: none;
        width: calc(100% - 5em);
        border: 1px solid #ddd;
        border-radius: 10px;
        height: 3em;
        padding: 0.6em;
    }

    .pseudoLoggedIn {
        width: 9px;
        height: 9px;
        border-radius: 5px;
        position: absolute;
        background-color: #95d731;
        left: 2.2em;
        top: 2.2em;
    }

    .buttonContainer {
        margin: 0.5em 1.5em 0 0;
    }
}