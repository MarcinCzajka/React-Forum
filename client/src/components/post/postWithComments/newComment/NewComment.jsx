import React from 'react';
import PropTypes from 'prop-types';
import { Form, Button } from 'semantic-ui-react';
import './NewComment.scss';

const NewComment = props => {
    return (
        <Form className='newCommentForm' reply>
            <Button 
                circular
                color='teal'
                size='massive'
                icon='comment alternate outline'
                onClick={props.changeReplyFormVisibility}
            />

            {props.active ? (
                <>
                    <Form.TextArea 
                        style={{marginTop: '0.5em'}}
                        value={props.replyContent}
                        onChange={props.handleReplyChange}
                    />

                    <Button 
                        onClick={props.handleReplyBtnClick}
                    />
                </>
            ) : ''}
        </Form>
    )
}

NewComment.defaultProps = {
    replyContent: '',
    active: false
}

NewComment.propTypes = {
    replyContent: PropTypes.string,
    active: PropTypes.bool,
    handleReplyChange: PropTypes.func.isRequired,
    handleReplyBtnClick: PropTypes.func.isRequired,
    changeReplyFormVisibility: PropTypes.func.isRequired
}

export default NewComment;