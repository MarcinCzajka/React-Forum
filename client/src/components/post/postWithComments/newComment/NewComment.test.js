import React from 'react';
import { create } from "react-test-renderer";
import { shallow } from 'enzyme';
import NewComment from './NewComment';

const emptyFnc = () => {};

const defaultProps = {
    handleReplyChange:emptyFnc,
    handleReplyBtnClick:emptyFnc,
    changeReplyFormVisibility:emptyFnc
}

describe('NewComment component', () => {
    test('renders without crashing', () => {
        shallow(<NewComment defaultProps />);
    });

    test('match snapshot w/o being active', () => {
        const component = create(<NewComment defaultProps />);
        expect(component).toMatchSnapshot();
    });

    test('match snapshot w/ being active', () => {
        const component = create(<NewComment active={true} defaultProps />);
        expect(component).toMatchSnapshot();
    });

    test('match snapshot w/ replyContent 1/2', () => {
        const component = create(<NewComment active={true} replyContent='123' defaultProps />);
        expect(component).toMatchSnapshot();
    });

    test('match snapshot w/ replyContent 1/2', () => {
        const component = create(<NewComment active={true} replyContent='asdfsdgdfgsd sdfsdf dsfdsgfdg dfdf' defaultProps />);
        expect(component).toMatchSnapshot();
    });

});