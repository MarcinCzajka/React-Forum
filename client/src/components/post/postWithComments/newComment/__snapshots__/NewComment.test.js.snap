// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewComment component match snapshot w/ being active 1`] = `
<form
  className="ui reply form newCommentForm"
  onSubmit={[Function]}
>
  <button
    className="ui teal massive circular icon button"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="comment alternate outline icon"
      onClick={[Function]}
    />
  </button>
  <div
    className="field"
  >
    <textarea
      onChange={[Function]}
      onInput={[Function]}
      rows={3}
      style={
        Object {
          "marginTop": "0.5em",
        }
      }
      value=""
    />
  </div>
  <button
    className="ui button"
    onClick={[Function]}
  />
</form>
`;

exports[`NewComment component match snapshot w/ replyContent 1/2 1`] = `
<form
  className="ui reply form newCommentForm"
  onSubmit={[Function]}
>
  <button
    className="ui teal massive circular icon button"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="comment alternate outline icon"
      onClick={[Function]}
    />
  </button>
  <div
    className="field"
  >
    <textarea
      onChange={[Function]}
      onInput={[Function]}
      rows={3}
      style={
        Object {
          "marginTop": "0.5em",
        }
      }
      value="123"
    />
  </div>
  <button
    className="ui button"
    onClick={[Function]}
  />
</form>
`;

exports[`NewComment component match snapshot w/ replyContent 1/2 2`] = `
<form
  className="ui reply form newCommentForm"
  onSubmit={[Function]}
>
  <button
    className="ui teal massive circular icon button"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="comment alternate outline icon"
      onClick={[Function]}
    />
  </button>
  <div
    className="field"
  >
    <textarea
      onChange={[Function]}
      onInput={[Function]}
      rows={3}
      style={
        Object {
          "marginTop": "0.5em",
        }
      }
      value="asdfsdgdfgsd sdfsdf dsfdsgfdg dfdf"
    />
  </div>
  <button
    className="ui button"
    onClick={[Function]}
  />
</form>
`;

exports[`NewComment component match snapshot w/o being active 1`] = `
<form
  className="ui reply form newCommentForm"
  onSubmit={[Function]}
>
  <button
    className="ui teal massive circular icon button"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="comment alternate outline icon"
      onClick={[Function]}
    />
  </button>
  
</form>
`;
